File: project/android/apps/MnnLlmApp/app/src/main/java/com/alibaba/mls/api/source/RepoConfig.java
// Created by ruoyi.sjd on 2025/2/11.
// Copyright (c) 2024 Alibaba Group Holding Limited All rights reserved.

package com.alibaba.mls.api.source;

public class RepoConfig {
    public String modelScopePath;
    public String huggingFacePath;
    public String modelId;

    public RepoConfig(String repositoryPath, String huggingFaceId, String modelId) {
        this.modelScopePath = repositoryPath;
        this.huggingFacePath = huggingFaceId;
        this.modelId = modelId;
    }

    public String repositoryPath() {
        return ModelSources.get().getRemoteSourceType() == ModelSources.ModelSourceType.HUGGING_FACE ? huggingFacePath : modelScopePath;
    }
}

File: apps/frameworks/model_downloader/android/src/main/java/com/alibaba/mls/api/source/RepoConfig.kt
// Created by ruoyi.sjd on 2025/2/11.
// Copyright (c) 2024 Alibaba Group Holding Limited All rights reserved.
package com.alibaba.mls.api.source
import com.alibaba.mls.api.source.ModelSources.Companion.get

class RepoConfig(var modelScopePath: String,
                 private var huggingFacePath: String,
                 var modelId: String) {
    fun repositoryPath(): String {
        return if (get().remoteSourceType == ModelSources.ModelSourceType.HUGGING_FACE)
            huggingFacePath
            else
            modelScopePath
    }
}